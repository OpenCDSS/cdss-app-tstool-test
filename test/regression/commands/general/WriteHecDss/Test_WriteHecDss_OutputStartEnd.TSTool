# Write Yearly data to DSS
# Test setup...
StartLog(LogFile="Results/Test_WriteHecDss_OutputStartEnd.TSTool.log")
# SetDebugLevel(LogFileLevel=10)
# RemoveFile(InputFile="Results\Test_WriteHecDss_Year_out.dss")
RemoveFile(InputFile="Results\Test_WriteHecDss_OutputStartEnd.dssutl.out",IfNotFound=Ignore)
# The generated time series has identifier parts that indicate the HEC-DSS
# parts, other than the D part, which is related to the period of record.
TS Alias1 = NewPatternTimeSeries(NewTSID="Apart1:Bpart1-with-dashes..Cpart1.Year.Fpart1",Description="Test data",SetStart="1950",SetEnd="1975",Units="MGD",PatternValues="5,10,12,13,75")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_OutputStartEnd_out.dss",OutputStart="1955",OutputEnd="1965",Type=PER-CUM)
WriteHecDss(OutputFile="Results/Test_WriteHecDss_OutputStartEnd_out.dss",OutputStart="1955",Type=PER-CUM,A="OnlyStart")
TS Alias2 = NewPatternTimeSeries(NewTSID="Apart2:Bpart2-with-dashes..Cpart2.Day.Fpart2",Description="Test data",SetStart="1950-01-01",SetEnd="1950-02-13",Units="MGD",PatternValues="5,10,12,13,75")
WriteHecDss(OutputFile="Results/Test_WriteHecDss_OutputStartEnd_out.dss",OutputStart="1950-01-15",OutputEnd="1950-01-28",TSList=AllMatchingTSID,TSID="Alias2",Type=PER-CUM)
#
# Use DSSUTL to output the data from the resulting DSS file to compare results
RunDSSUTL(DssFile="Results\Test_WriteHecDss_OutputStartEnd_out.dss",InputFile="Test_WriteHecDss_OutputStartEnd.MCO",DssutlProgram="${WorkingDir}\DSSUTL.EXE")
# In order to regenerate the expected results file, edit the output file in the file: Test_WriteHecDss_OutputStartEnd.MCO
CompareFiles(InputFile1="ExpectedResults\Test_WriteHecDss_OutputStartEnd.dssutl.out",InputFile2="Results/Test_WriteHecDss_OutputStartEnd.dssutl.out",WarnIfDifferent=True)
