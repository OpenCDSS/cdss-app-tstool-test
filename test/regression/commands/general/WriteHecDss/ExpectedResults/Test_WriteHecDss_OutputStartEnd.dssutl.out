  
 /APART1/BPART1-WITH-DASHES/CPART1/01JAN1900/1YEAR/FPART1/
 Start: 31DEC1950 at 2400 hours;   End: 31DEC1960 at 2400 hours;  Number:   11
 Units: MGD        Type: PER-CUM 
 31DEC1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31DEC1957, 2400;    12.00   13.00   75.00    5.00
 ----------------------------------------------------------------------------
  
 /APART2/BPART2-WITH-DASHES/CPART2/01JAN1950/1DAY/FPART2/
 Start: 01JAN1950 at 2400 hours;   End: 14JAN1950 at 2400 hours;  Number:   14
 Units: MGD        Type: PER-CUM 
 01JAN1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 08JAN1950, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 ----------------------------------------------------------------------------
  
 /ONLYSTART/BPART1-WITH-DASHES/CPART1/01JAN1900/1YEAR/FPART1/
 Start: 31DEC1950 at 2400 hours;   End: 31DEC1970 at 2400 hours;  Number:   21
 Units: MGD        Type: PER-CUM 
 31DEC1950, 2400;     5.00   10.00   12.00   13.00   75.00    5.00   10.00
 31DEC1957, 2400;    12.00   13.00   75.00    5.00   10.00   12.00   13.00
 31DEC1964, 2400;    75.00    5.00   10.00   12.00   13.00   75.00    5.00
 ----------------------------------------------------------------------------
  
